### PR TITLE
Add support for CircleCI

### PR DESCRIPTION
**Pull request by sjackman**
This PR enables building bottles on CircleCI.

### CODE REVIEW

File: cmd/brew-test-bot.rb [hunk 1]
```diff
@@ -1548,7 +1554,8 @@ def sanitize_argv_and_env
                              "#{HOMEBREW_PREFIX}/bin:#{HOMEBREW_PREFIX}/sbin:#{ENV["PATH"]}"
 
     travis = !ENV["TRAVIS"].nil?
-    if travis
+    circle = !ENV["CIRCLECI"].nil?
+    if travis || circle
       ARGV << "--verbose" << "--ci-auto" << "--no-pull"
       ENV["HOMEBREW_TRAVIS_CI"] = "1"
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: This shouldn't be set on `circle`

File: cmd/brew-test-bot.rb [hunk 2]
```diff
@@ -1548,7 +1554,8 @@ def sanitize_argv_and_env
...
       ENV["HOMEBREW_TRAVIS_SUDO"] = ENV["TRAVIS_SUDO"]
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: This shouldn't be set on `circle`

**Comment by sjackman** [comment]
Good eye. Fixed.

File: cmd/brew-test-bot.rb [hunk 3]
```diff
@@ -1548,13 +1554,16 @@ def sanitize_argv_and_env
                              "#{HOMEBREW_PREFIX}/bin:#{HOMEBREW_PREFIX}/sbin:#{ENV["PATH"]}"
 
     travis = !ENV["TRAVIS"].nil?
-    if travis
+    circle = !ENV["CIRCLECI"].nil?
+    if travis || circle
       ARGV << "--verbose" << "--ci-auto" << "--no-pull"
-      ENV["HOMEBREW_TRAVIS_CI"] = "1"
-      ENV["HOMEBREW_TRAVIS_SUDO"] = ENV["TRAVIS_SUDO"]
       ENV["HOMEBREW_COLOR"] = "1"
```
[the remark this answers is not in the document]
Comment (sjackman) [reply cmd/brew-test-bot.rb]: No. Yipes. I shouldn't submit PRs at 1:30 AM.

File: cmd/brew-test-bot.rb [hunk 4]
```diff
@@ -1592,11 +1601,12 @@ def sanitize_argv_and_env
     jenkins_branch = !ENV["GIT_COMMIT"].nil?
     jenkins_branch ||= jenkins_pipeline_branch
     azure_pipelines_pr = ENV["BUILD_REASON"] == "PullRequest"
+    circle_pr = !ENV["CIRCLE_PULL_REQUEST"].to_s.empty?
```
[the remark this answers is not in the document]
Comment (sjackman) [reply cmd/brew-test-bot.rb]: The code base uses `empty?` 19 times and `blank?` 0 times, so I'm inclined to go with `empty?` for consistency. It shouldn't make a practical difference.

File: cmd/brew-test-bot.rb [hunk 5]
```diff
@@ -415,8 +415,8 @@ def download
           @hash = nil
         end
       # Use Circle CI pull-request variables for pull request jobs.
-      elsif !ENV["CI_PULL_REQUEST"].to_s.empty?
-        @url = ENV["CI_PULL_REQUEST"]
+      elsif !ENV["CIRCLE_PULL_REQUEST"].to_s.empty?
```
[the remark this answers is not in the document]
Comment (sjackman) [reply cmd/brew-test-bot.rb]: Ditto.

**Comment by sjackman** [comment]
Thanks for the review, Mike! You caught a few important gafs.

**Pull request merged**